packages/zowe-explorer/src/commands/UnixCommandHandler.ts

```diff
@@ -43,219 +38,181 @@ export class UnixCommandHandler extends ZoweCommandProvider {
+                    vscode.l10n.t("An SSH profile will be used for issuing UNIX commands with the profile {0}.", [this.nodeProfile?.name])
                 );
             } catch (e) {
                 // error would be due to missing API, assuming SSH profile not required
                 ZoweLogger.warn(
                     vscode.l10n.t(
                         "Error checking if SSH profile type required for issuing UNIX commands, setting requirement to false for profile {0}.",
-                        [this.serviceProf?.name]
+                        [this.nodeProfile?.name]
                     )
                 );
             }
-            await this.profileInstance.checkCurrentProfile(this.serviceProf);
+            await this.profileInstance.checkCurrentProfile(this.nodeProfile);
 
             if (this.isSshRequiredForProf) {
-                await this.setsshSession();
+                await this.getSshProfile();
+                if (!this.sshProfile) {
+                    return;
+                }
+
+                const cmdArgs: imperative.ICommandArguments = this.getSshCmdArgs(this.sshProfile.profile);
+                // create the ssh session
+                const sshSessCfg = zosuss.SshSession.createSshSessCfgFromArgs(cmdArgs);
+                imperative.ConnectionPropsForSessCfg.resolveSessCfgProps<zosuss.ISshSession>(sshSessCfg, cmdArgs);
+                this.sshSession = new zosuss.SshSession(sshSessCfg);
+
                 if (!this.sshSession) {
-                    this.serviceProf = undefined;
+                    this.nodeProfile = undefined;
                     ZoweLogger.error(this.unixCmdMsgs.sshSessionErrorMsg);
                     Gui.errorMessage(this.unixCmdMsgs.sshSessionErrorMsg);
                     return;
                 }
             }
 
-            if (!cwd) {
+            await this.profileInstance.checkCurrentProfile(this.nodeProfile);
+
+            if (!this.sshCwd) {
                 const options: vscode.InputBoxOptions = {
                     prompt: vscode.l10n.t("Enter the path of the directory in order to execute the command"),
                 };
-                cwd = await Gui.showInputBox(options);
-                if (cwd === "") {
+                this.sshCwd = await Gui.showInputBox(options);
+                if (this.sshCwd === "") {
                     ZoweLogger.info(this.unixCmdMsgs.cwdRedirectingMsg);
                     Gui.showMessage(this.unixCmdMsgs.cwdRedirectingMsg);
                 }
-                if (cwd == undefined) {
-                    this.serviceProf = undefined;
-                    ZoweLogger.info(this.unixCmdMsgs.opCancelledMsg);
-                    Gui.showMessage(this.unixCmdMsgs.opCancelledMsg);
+                if (this.sshCwd == undefined) {
+                    this.nodeProfile = undefined;
+                    ZoweLogger.info(this.operationCancelled);
+                    Gui.showMessage(this.operationCancelled);
```

Comment:
Although I'm not explicitly requesting changes for this, it'd be nice to get rid of this "Operation cancelled" message as well 😋